packages/maskbook/src/extension/options-page/DashboardComponents/FixedTokenList.tsx

```diff
@@ -42,18 +47,43 @@ export function FixedTokenList(props: FixedTokenListProps) {
         FixedSizeListProps,
     } = props
 
-    //#region search tokens
     const { ERC20_TOKEN_LISTS } = useEthereumConstants()
     const [address, setAddress] = useState('')
+    const { MASK_ADDRESS } = useTokenConstants()
+
     const { state, tokensDetailed: erc20TokensDetailed } = useERC20TokensDetailedFromTokenLists(
         ERC20_TOKEN_LISTS,
         keyword,
     )
-    //#endregion
 
-    //#region mask token
-    const { MASK_ADDRESS } = useTokenConstants()
-    //#endregion
+    const filteredTokens = erc20TokensDetailed.filter(
+        (token) =>
+            (!includeTokens.length || includeTokens.some(currySameAddress(token.address))) &&
+            (!excludeTokens.length || !excludeTokens.some(currySameAddress(token.address))),
+    )
+
+    const renderTokens = uniqBy([...tokens, ...filteredTokens], (x) => x.address.toLowerCase()).sort((a, z) => {
+        if (a.type === EthereumTokenType.Native) return -1
+        if (z.type === EthereumTokenType.Native) return 1
+        if (isSameAddress(a.address, MASK_ADDRESS ?? '')) return -1
+        if (isSameAddress(z.address, MASK_ADDRESS ?? '')) return 1
+        return 0
+    })
+
+    const {
+        loading: loadingAssets,
+        value: assets,
+        error,
+    } = useAssetsFromChain(renderTokens.filter((x) => EthereumAddress.isValid(x.address)))
+
+    const renderAssets =
+        error || !account || loadingAssets
+            ? renderTokens.map((token) => ({ token: token, balance: null }))
+            : assets.sort((a, b) => {
```

Comment:
Actually, we've got a bug here. The demand is to render the MASK token in the first place. But we sorted twice here. The second one will override the previous result. Could you merge them into one sorting?

Reply:
OK